File: app/src/main/java/com/ivantha/ihaveatheory/util/Controller.java
package com.ivantha.ihaveatheory.util;

import android.content.Context;
import android.content.Intent;

import com.avoid.ihaveatheory.game.situation.Situation;
import com.ivantha.ihaveatheory.activity.ChoiceActivity;
import com.ivantha.ihaveatheory.activity.ProgressActivity;
import com.ivantha.ihaveatheory.activity.SituationActivity;
import com.ivantha.ihaveatheory.game.choice.Choice;
import com.ivantha.ihaveatheory.game.scene.Scene;

public class Controller {
    public static void showProgress(Context context, String title, String time) {
        Intent intent = new Intent(context, ProgressActivity.class);
        intent.putExtra("title", title);
        intent.putExtra("time", time);
        context.startActivity(intent);
    }

    public static void showChoice(Context context, Choice choice) {
        Intent intent = new Intent(context, ChoiceActivity.class);
        intent.putExtra("choice", choice);
        context.startActivity(intent);
    }

    public static void showSituation(Context context, Situation situation){
        Intent intent = new Intent(context, SituationActivity.class);
        intent.putExtra("situation", situation);
        context.startActivity(intent);
    }


    public static void showScene(Context context, Scene scene){

    }
}

File: app/src/main/java/com/avoid/ihaveatheory/util/Controller.kt
package com.avoid.ihaveatheory.util

import android.content.Context
import android.content.Intent
import com.avoid.ihaveatheory.activity.ChoiceActivity
import com.avoid.ihaveatheory.activity.ProgressActivity
import com.avoid.ihaveatheory.activity.SituationActivity
import com.avoid.ihaveatheory.game.choice.Choice
import com.avoid.ihaveatheory.game.scene.Scene
import com.avoid.ihaveatheory.game.situation.Situation

object Controller {
    fun showProgress(context: Context, title: String, time: String) {
        val intent = Intent(context, ProgressActivity::class.java)
        intent.putExtra("title", title)
        intent.putExtra("time", time)
        context.startActivity(intent)
    }

    fun showChoice(context: Context, choice: Choice) {
        val intent = Intent(context, ChoiceActivity::class.java)
        intent.putExtra("choice", choice)
        context.startActivity(intent)
    }

    fun showSituation(context: Context, situation: Situation) {
        val intent = Intent(context, SituationActivity::class.java)
        intent.putExtra("situation", situation)
        context.startActivity(intent)
    }


    fun showScene(context: Context, scene: Scene) {

    }
}